internal/service/rds/cluster.go

```diff
@@ -1394,6 +1394,24 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, meta int
 		}
 	}
 
+	if d.HasChange("replication_source_identifier") {
```

Comment:
If you change this to a different value, should it be `ForceNew`?

Reply:
Agreed that it should do something. The current behavior is to "do nothing", which is an oversight.